app/screens/end-normal-game.js

```diff
@@ -16,159 +31,196 @@ class EndNormalGame extends React.PureComponent {
     };
     this.setModalSignIn = this.setModalSignIn.bind(this);
     this.setModalSignUp = this.setModalSignUp.bind(this);
-    this.setModalHide = this.setModalHide.bind(this);
+    this.setModalHide = this.setModalHide.bind(this);  //si uso el this tengo que bindear
+    this.renderLeaderBoard = this.renderLeaderBoard.bind(this);
+    this.saveMatch = this.saveMatch.bind(this);
   }
 
   setModalSignIn() {
-    this.setState({ showModal: 'signIn' });    
+    this.setState({ showModal: 'signIn' });
   }
 
   setModalSignUp() {
     this.setState({ showModal: 'signUp' });
   }
 
   setModalHide() {
-    this.setState({ showModal: 'hide' });    
+    this.setState({ showModal: 'hide' });  
+  }
+
+  saveMatch() {
+    matchService.update(this.props.matchData.match)
+      .then((res) => {
+        this.props.history.push('/');
+      })
+      .catch((err) => {
+        alert(err.error);
+      });
+  }
+
+  renderLeaderBoard() {
+    var init_leaderboard = this.props.matchData.match.game.ranking.slice(); //clean copy of ranking
+    init_leaderboard.push({ user: this.props.matchData.state.player,
+                       points: this.props.matchData.state.score
+                    });
```

Comment:
arreglar indentacion:
    init_leaderboard.push({ 
    __user: this.props.matchData.state.player, 
    __points: this.props.matchData.state.score 
    });
(los __ no van, los puse porque el comentario ponia todo en el mismo nivel, es para mostrar los dos espacios)